WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -32,6 +33,23 @@ public CoinJoinManager(WalletManager walletManager, RoundStateUpdater roundStatu
 		public IWabiSabiApiRequestHandler ArenaRequestHandler { get; }
 		public RoundStateUpdater RoundStatusUpdater { get; }
 		public ServiceConfiguration ServiceConfiguration { get; }
+		private ImmutableDictionary<string, WalletTrackingData> TrackedWallets { get; set; } = ImmutableDictionary<string, WalletTrackingData>.Empty;
+
+		public CoinJoinClientState HighestCoinJoinClientState
+		{
+			get
+			{
+				var coinjoinClients = TrackedWallets.Values.GroupBy(wtd => wtd.CoinJoinClient.State).ToImmutableArray();
```

Comment:
Here I would filter for those coinjoin tasks that are running and ignore those that are already finished (no matter how or why). Perhaps something like:

```c#
var runningCoinjoins = trackedWallets.Where(x => !x.Value.CoinJoinTask.IsCompleted);
if (!runningCoinjoins.Any())
{
    return idle;
}
else if(runningCoinjoins.Any(x => x.State == critial)
{ retuyn crtitical;
}
else
return inprogress;

Reply:
ACK